### PR TITLE
exporters/datadogexporter: update and improve examples

### PR DESCRIPTION
**Pull request by gbbr**
This change deletes the old examples which were no longer working, were
based on deprecated values and used older APIs. It replaces them with
newer up-to-date ones which are tested and known to work.

Added:
* example/collector.yaml: an example configuration yaml for the collector, set up
with Datadog, containing all current options along with their documentation.
* example/k8s-chart: contains a chart which sets up the collector in a deployment,
ready to collect traces and add K8S attribute.

In a follow-up PR, a README.md should be provided to better explain the
purpose of these.

### CODE REVIEW

File: exporter/datadogexporter/examples/collector.yaml [hunk 1]
```diff
@@ -1,63 +1,41 @@
...
+  # https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/datadogexporter
   datadog:
     ## @param hostname - string - optional
     ## A custom hostname.
-    ## If unset it will be determined from the `DD_HOST` environment variable (Deprecated: [v0.47.0] set environment variable explicitly on configuration instead).
     ## If both this and `DD_HOST` are unset, this will be determined automatically if possible.
     #
     # hostname: customhostname
 
-    ## @param env - string - optional
```
Comment (mx-psi) [inline exporter/datadogexporter/examples/collector.yaml:16]: Don't remove the deprecated settings for now, I will do that on #12110

File: exporter/datadogexporter/examples/sdk-go/go.mod [hunk 2]
```diff
@@ -0,0 +1,31 @@
+module sdk-example
```
Comment (mx-psi) [inline exporter/datadogexporter/examples/sdk-go/go.mod:1]: Anything other than Collector YAML configuration samples should not be in this repository (see #7155 and #11074 for precedent)

File: exporter/datadogexporter/examples/k8s-agent/configmap.yaml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        timeout: 5s
+        override: false
+      k8sattributes:
+        # associate to pods using the k8s.pod.uid resource attribute
+        # set by each pod's template
+        pod_association:
+          - from: resource_attribute
+            name: k8s.pod.uid
+      batch:
+        timeout: 10s
```
Comment (mx-psi) [inline exporter/datadogexporter/examples/k8s-agent/configmap.yaml:33]:
```suggestion
```

**Review by mx-psi — CHANGES_REQUESTED** [review]

File: exporter/datadogexporter/examples/collector.yaml [hunk 4]
```diff
@@ -185,11 +208,6 @@ exporters:
     ## Trace exporter specific configuration.
     #
     # traces:
-      ## @param max_events_per_second - integer - optional - default: 200
```
Comment (gbbr) [inline exporter/datadogexporter/examples/collector.yaml:188]: This option doesn't exist in the code. Not sure why it is here.

File: exporter/datadogexporter/examples/k8s-agent/configmap.yaml (hunk 3 above)
Comment (mx-psi) [reply exporter/datadogexporter/examples/k8s-agent/configmap.yaml:33]: Okay, let's keep it this way then

File: exporter/datadogexporter/examples/k8s-agent/deployment.yaml [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  selector:
+    matchLabels:
+      app: opentelemetry
+      component: otel-collector
+  template:
+    metadata:
+      labels:
+        app: opentelemetry
+        component: otel-collector
+    spec:
```
Comment (dineshg13) [inline exporter/datadogexporter/examples/k8s-agent/deployment.yaml:18]: Could change the name & labels from `otel-collector` to `app` ?

File: exporter/datadogexporter/examples/collector.yaml [hunk 6]
```diff
@@ -255,10 +273,25 @@ exporters:
...
       # tags: []
 
-
+# `service` defines the Collector pipelines, observability settings and extensions.
 service:
+  # `telemetry` allows you to customize the Collector's own telemetry.
+  telemetry:
+    logs:
+      # Set the log level to info (the default)
+      level: info
```
Comment (mx-psi) [inline exporter/datadogexporter/examples/collector.yaml:282]: I would remove this since this is the default
```suggestion
```

Comment (gbbr) [reply exporter/datadogexporter/examples/collector.yaml:282]: Agreed. There is no gain in documenting OTel options. The purpose here should be to underline Datadog exporter config, without any extra noise.

**Review by mx-psi — COMMENTED** [review]
Lint is failing because of a typo; otherwise this LGTM

File: exporter/datadogexporter/examples/k8s-agent/service.yaml [hunk 7]
```diff
@@ -0,0 +1,5 @@
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: otel-collector-account
+  namespace: default
```
Comment (dineshg13) [inline exporter/datadogexporter/examples/k8s-agent/service.yaml:5]: Could you name this file serviceaccount.yaml ?

Comment (gbbr) [reply exporter/datadogexporter/examples/k8s-agent/service.yaml:5]: Yes! Should I just move it to `roles.yaml`? Is that going to be better? I feel it's perhaps not worth its own file.

Comment (gbbr) [reply exporter/datadogexporter/examples/k8s-agent/service.yaml]: Done!

**Comment by mx-psi** [comment]
Lint is still failing:
```
 running misspell -error
/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/exporter/datadogexporter/examples/k8s-chart/deployment.yaml:29:79: "succesfully" is a misspelling of "successfully"
```

**Comment by mx-psi** [comment]
Merging #12110 caused a merge conflict here, could you fix it? Hopefully git is smart enough to make this smooth...

File: exporter/datadogexporter/examples/k8s-chart/configmap.yaml [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+        detectors: [system, env, docker]
+        timeout: 5s
+        override: false
+      # adds various tags related to k8s
+      k8sattributes:
+      batch:
+        timeout: 10s
+    service:
+      pipelines:
+        traces:
```
Comment (dineshg13) [inline exporter/datadogexporter/examples/k8s-chart/configmap.yaml:32]: can u add metrics as well here ?

Comment (gbbr) [reply exporter/datadogexporter/examples/k8s-chart/configmap.yaml:36]: Good point. Added.

**Review by mx-psi — APPROVED** [review]
LGTM (but will wait until @dineshg13 approves as well to merge)

**Review by dineshg13 — APPROVED** [review]
LTGM

**Pull request merged**